cfme/tests/integration/test_menu_plugin.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+import cfme.configure.access_control as ac
+import fauxfactory
+from cfme import Credential, login
+from utils import menu_plugin as mp
+from cfme.fixtures import pytest_selenium as sel
+from utils.providers import setup_a_provider
+
+
+@pytest.fixture(scope="module")
+def setup_infra_provider():
+    return setup_a_provider(prov_class="infra", validate=True, check_existing=True,
+        required_keys=['ownership_vm'])
+
+
+@pytest.fixture(scope="module")
+def setup_menu_plugin():
+    mp.menu_plugin_setup()
+
+
+@pytest.yield_fixture(scope="module")
+def create_new_role():
+    login.login_admin()
```

Comment:
Why do we need so many ```login_admin()``` calls? 
* We should only have needed to do this when we are logging in as a non-admin user
* The context manager used below ```with non_admin_user``` should logout for us. Logging in again should only be necessary if the next call you try to make doesn't use a sel.force_navigate